src/components/connections/networks-helpers.js

```diff
@@ -0,0 +1,229 @@
+      case 'main':
+        mainNetworks.appendChild(networkItem);
+        break;
+      case 'test':
+        testNetworks.appendChild(networkItem);
+        break;
+      default:
+        break;
+    }
+  });
+}
+
+export function createNetworkItem(network) {
+  const item = document.createElement('div');
+  item.className = 'network-modal-item';
+  item.dataset.chainId = network.chainId;
+  item.innerHTML = `
+    <div class="network-modal-item-content">
+      <div class="network-modal-item-icon" style="background-color: ${network.color}"></div>
+      <div class="network-modal-item-info">
+        <div class="network-modal-item-name">${network.name}</div>
+        <div class="network-modal-item-chain-id">${network.chainId}</div>
+      </div>
+    </div>
+  `;
+
+  item.addEventListener('click', async () => {
+    hideNetworkError(); // Hide any existing error before attempting to switch
+    await switchNetwork(network.chainId);
+    document.querySelector('.network-modal').style.display = 'none';
+  });
+
+  return item;
+}
+
+export function showNetworkError(message) {
+  // Remove any existing error message
+  hideNetworkError();
+
+  // Create error message element
+  const errorDiv = document.createElement('div');
+  errorDiv.id = 'networkError';
+  errorDiv.className = 'error-message';
+  errorDiv.style.marginTop = '10px';
+  errorDiv.style.width = '100%';
+  errorDiv.innerHTML = `<div class="error-message-text">${message}</div>`;
+
+  // Find the network picker button and insert error after it
+  const networkButton = document.getElementById('openNetworkPicker');
+  const cardBody = networkButton.closest('.card-body');
+  cardBody.appendChild(errorDiv);
+
+  // Auto-hide after 5 seconds
+  setTimeout(() => {
+    hideNetworkError();
+  }, 5000);
+}
+
+export function hideNetworkError() {
+  const existingError = document.getElementById('networkError');
+  if (existingError) {
+    existingError.remove();
+  }
+}
+
+export async function switchNetwork(chainId) {
+  if (!globalContext.provider) {
+    console.error('No provider available');
+    return;
+  }
+
+  try {
+    await globalContext.provider.request({
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId }],
+    });
+  } catch (switchError) {
+    // This error code indicates that the chain has not been added to MetaMask.
+    if (switchError.code === 4902) {
+      const network = NETWORKS.find((n) => n.chainId === chainId);
+      const networkName = network ? network.name : `Chain ID ${chainId}`;
+      showNetworkError(`${networkName} is not available in your wallet`);
+    } else {
+      console.error('Error switching network:', switchError);
+      showNetworkError('Failed to switch network');
+    }
+  }
+}
+
+export function updateCurrentNetworkDisplay() {
+  const currentNetworkName = document.getElementById('currentNetworkName');
+
+  if (
+    globalContext.chainIdInt !== undefined &&
+    globalContext.chainIdInt !== null
+  ) {
```

Comment:
```suggestion
if (globalContext.chainIdInt) {
```
from what I understand, we don't care what type of value this is, as long as it's defined, so
this condition can be simplified